src/gui/DatabaseTabWidget.cpp

```diff
@@ -128,20 +128,18 @@ void DatabaseTabWidget::openDatabase(const QString& fileName, const QString& pw,
     DatabaseManagerStruct dbStruct;
 
     // test if we can read/write or read the file
-    QFile file(fileName);
     // TODO: error handling
-    if (!file.open(QIODevice::ReadWrite)) {
-        if (!file.open(QIODevice::ReadOnly)) {
+    if (!fileInfo.isReadable() && !fileInfo.isWritable()) {
+        if (fileInfo.isReadable()) {
```

Comment:
You are right.
isReadable()/isWriteable() only check  the permission of the file.
It will not work in read-only filesystem situation.